volatility/framework/plugins/windows/poolscanner.py

```diff
@@ -445,13 +448,19 @@ def _get_pool_header_module(cls, context, layer_name, symbol_table):
             else:
                 pool_header_json_filename = "poolheader-x86"
 
+            is_vista_or_later = cls.is_vista_or_later(context, symbol_table)
+            if is_vista_or_later:
+                class_type = extensions.pool.POOL_HEADER_VISTA
+            else:
+                class_type = extensions.pool.POOL_HEADER
+
             new_table_name = intermed.IntermediateSymbolTable.create(
                 context = context,
                 config_path = configuration.path_join(context.symbol_space[symbol_table].config_path, "poolheader"),
                 sub_path = "windows",
                 filename = pool_header_json_filename,
                 table_mapping = {'nt_symbols': symbol_table},
-                class_types = {'_POOL_HEADER': extensions.pool.POOL_HEADER})
+                class_types = {'_POOL_HEADER': class_type})
```

Comment:
This seems like it only picks up one very specific place that this is used (poolscanner) and the logic wouldn't be applied to normal windows symbol tables.  I think the logic should also be applied at https://github.com/volatilityfoundation/volatility3/blob/master/volatility/framework/symbols/windows/__init__.py#L39 which means mebbe moving the decision to a function that lives somewhere under `framework/extensions/windows` (mebbe `extensions/pool.py`?) and then calling it from the poolscanner...

Reply:
Either that or writing the class to do the decision when it runs, but I think that'll cause more issues in the longer term...

Reply:
Yeah, that makes sense.  This only happens in the backup case when the symbols aren't found in the kernel pdb.  Not sure if we can determine the OS version at the point in the code? Will look later.